elastic/elastic-agent
# Update make check-ci command to support kustomize manifest updates

- Enhancement

## What does this PR do?

Updates the make file of elastic agent with new command `@GENERATEKUSTOMIZE=true $(MAKE) -C deploy/kubernetes generate-k8s`.
This command generates the kustomize manifests for deploying elastic agent

## Why is it important?

- https://github.com/elastic/elastic-agent/pull/5056#issuecomment-2214267966

Kustomize templates have been added in this [PR](- https://github.com/elastic/elastic-agent/pull/4754) in order to introduce another flow to onboard users into Observability offering.

## Checklist


## How to test this PR locally

- Download this branch
- Change version of agent in https://github.com/elastic/elastic-agent/blob/main/version/version.go#L7, to an example value  eg defaultBeatVersion = "8.18.0"
- commit change:  `git commit -m "some msg"`, git push
- Run command `make check-ci`
- See that the following files should be updated
```bash
git status

 modified:   deploy/kubernetes/elastic-agent-kustomize/default/elastic-agent-managed/base/elastic-agent-managed-daemonset.yaml
        modified:   deploy/kubernetes/elastic-agent-kustomize/default/elastic-agent-standalone/base/elastic-agent-standalone-daemonset.yaml
        modified:   deploy/kubernetes/elastic-agent-kustomize/ksm-autosharding/elastic-agent-managed/base/elastic-agent-managed-daemonset.yaml
        modified:   deploy/kubernetes/elastic-agent-kustomize/ksm-autosharding/elastic-agent-managed/extra/elastic-agent-managed-statefulset.yaml
        modified:   deploy/kubernetes/elastic-agent-kustomize/ksm-autosharding/elastic-agent-standalone/base/elastic-agent-standalone-daemonset.yaml
        modified:   deploy/kubernetes/elastic-agent-kustomize/ksm-autosharding/elastic-agent-standalone/extra/elastic-agent-standalone-statefulset.yaml
        modified:   deploy/kubernetes/elastic-agent-managed-kubernetes.yaml
        modified:   deploy/kubernetes/elastic-agent-standalone-kubernetes.yaml
```

## Related issues

- Relates https://github.com/elastic/elastic-agent/pull/5056#issuecomment-2214267966
- Relates https://github.com/elastic/elastic-agent/pull/4754
